### PR TITLE
Log Support in Attribute Processor (2/2) 

### PR DESCRIPTION
**Pull request by zeitlinger**
Contains merge for #1830

extract common tests from span and log filtering

### CODE REVIEW

**Review by keitwb — APPROVED** [review]
LGTM, thanks for doing this @zeitlinger.

**Comment by tigrannajaryan** [comment]
@zeitlinger can you please rebase from master so that changes from 1830 are not visible here?

**Comment by zeitlinger** [comment]
@tigrannajaryan rebase done

File: exporter/exportertest/nop_exporter.go [hunk 1]
```diff
@@ -45,19 +45,19 @@ func (ne *nopExporter) Shutdown(context.Context) error {
 }
 
 // NewNopTraceExporter creates an TraceExporter that just drops the received data.
-// Deprecated: Use consumertest.NewNopTraces.
+// Deprecated: Use consumertest.NewNopTraceExporter.
```
Comment (tigrannajaryan) [inline exporter/exportertest/nop_exporter.go:48]: Looks like we some unrelated changes here.

**Comment by tigrannajaryan** [comment]
@zeitlinger Can you please split the PR into 2 parts: the refactoring of the common tests for spans and logs and addition of logs support to attributes processor. This PR is big and difficult to follow because it moves large volumes of code between files and it is hard to see what is changed in the behavior.

File: exporter/exportertest/nop_exporter.go (hunk 1 above)
Comment (zeitlinger) [reply exporter/exportertest/nop_exporter.go:48]: yes, true - I noticed this while fixing the rebase conflicts - should this be a seprate PR?

**Comment by zeitlinger** [comment]
> @zeitlinger Can you please split the PR into 2 parts: the refactoring of the common tests for spans and logs and addition of logs support to attributes processor. This PR is big and difficult to follow because it moves large volumes of code between files and it is hard to see what is changed in the behavior.

will do

File: exporter/exportertest/nop_exporter.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/exportertest/nop_exporter.go:48]: Would be better, also the suggestion is wrong

Comment (zeitlinger) [reply exporter/exportertest/nop_exporter.go]: ok, dropped the change

**Review by tigrannajaryan — APPROVED** [review]
Thank you @zeitlinger ! Nice PR and highly wanted feature addition.
